tests/Microsoft.Identity.Test.Integration/HeadlessTests/B2CUsernamePasswordIntegrationTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License.
+
+using System.Net;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    // Note: these tests require permission to a KeyVault Microsoft account; 
+    // Please ignore them if you are not a Microsoft FTE, they will run as part of the CI build
+    [TestClass]
+    public class B2CUsernamePasswordIntegrationTests
+    {
+        private const string _b2CROPCAuthority = "https://msidlabb2c.b2clogin.com/tfp/msidlabb2c.onmicrosoft.com/B2C_1_ROPC_Auth";
+
+        private static readonly string[] s_b2cScopes = { "https://msidlabb2c.onmicrosoft.com/msidlabb2capi/read" };
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext context)
+        {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+        }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+        }
+
+        [TestMethod]
+        public async Task ROPC_B2C_Async()
+        {
+            var labResponse = LabUserHelper.GetB2CLocalAccount();
+            await RunB2CHappyPathTestAsync(labResponse).ConfigureAwait(false);
+        }
+
+        private async Task RunB2CHappyPathTestAsync(LabResponse labResponse)
+        {
+            var user = labResponse.User;
+
+            SecureString securePassword = new NetworkCredential("", user.Password).SecurePassword;
+
+            var msalPublicClient = PublicClientApplicationBuilder.Create("e3b9ad76-9763-4827-b088-80c7a7888f79").WithB2CAuthority(_b2CROPCAuthority).Build();
```

Comment:
getting the wrong client_id back from the lab so hard coded this for now, but will change before merging.